src/index.js

```diff
@@ -1,42 +1,105 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
+  let price = product.querySelector('.price span').innerHTML;
+  let quantity = product.querySelector('.quantity input').value;
+
+  let total = price * quantity;
+  let subtotal = product.querySelector('.subtotal span');
+
+  subtotal.innerHTML = total;
+  console.log(total)
   console.log('Calculating subtotal, yey!');
 
-  //... your code goes here
+  return total;
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
+  /* // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
   const singleProduct = document.querySelector('.product');
   updateSubtotal(singleProduct);
-  // end of test
+  // end of test */
 
   // ITERATION 2
-  //... your code goes here
 
-  // ITERATION 3
-  //... your code goes here
+  const totalProducts = document.getElementsByClassName('product');
+
+  let total = 0;
+  for(let i = 0; i < totalProducts.length; i++) {
+    total += updateSubtotal(totalProducts[i]);
+  } 
+
+  let showTotal = document.querySelector('#total-value span');
+  showTotal.innerHTML = total
+
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
   console.log('The target in remove is:', target);
-  //... your code goes here
+
+  const deleteRow = target.parentNode.parentNode;
+  const parentRow = target.parentNode.parentNode.parentNode;
```

Comment:
Another way to do this is using the method remove, like:

`target.parentNode.parentNode.remove();`